aten/src/ATen/native/ReduceOps.cpp

```diff
@@ -815,13 +886,26 @@ Tensor &any_out(Tensor &result, const Tensor &self, int64_t dim, bool keepdim) {
               "any only supports CPU AND CUDA device type, got: ", self.device().type());
   TORCH_CHECK(self.layout() == Layout::Strided,
               "any only supports strided layout, got: ", self.layout());
+  // Refer [all, any : uint8 compatibility]
+  TORCH_CHECK(result.scalar_type() == ScalarType::Bool || result.scalar_type() == ScalarType::Byte,
```

Comment:
This needs a slight update.

I suggest separating these lines into two `TORCH_CHECK`s. The first one checks if result's dtype is bool OR self's dtype is byte. If that check fails then the error message should report that any "expected" a boolean 'out' tensor but got ...

The second `TORCH_CHECK` checks that result's dtype is byte or self's dtype is not byte. If that check fails then the error message should report that any "expected" an int8 'out' tensor but got ...

These conditionals can be phrased slightly differently if you prefer.